corehq/apps/sso/tests/test_backends.py

```diff
@@ -48,26 +51,19 @@ def setUpClass(cls):
     def tearDownClass(cls):
         AuthenticatedEmailDomain.objects.all().delete()
         IdentityProvider.objects.all().delete()
-        cls.user.delete(cls.domain.name, deleted_by=None)
-
-        # cleanup "new" users
-        for username in [
-            'm@vaultwax.com',
-            'isa@vaultwax.com',
-            'zee@vaultwax.com',
-            'exist@vaultwax.com',
-            'aart@vaultwax.com',
-            'liam@vaultwax.com',
-            'nile@vaultwax.com'
-        ]:
-            web_user = WebUser.get_by_username(username)
-            if web_user:
-                web_user.delete(cls.domain.name, deleted_by=None)
-
-        cls.domain.delete()
-        cls.account.delete()
         super().tearDownClass()
 
+    def authenticate(self, request, **kw):
+        def delete(user):
+            if user is not None and user.username != self.user.username:
+                if not isinstance(user, WebUser):
+                    user = WebUser.get_by_username(user.username)
+                user.delete(self.domain.name, deleted_by=None)
```

Comment:
Do we only delete `WebUser` here? and still keep `Django User`?

Reply:
The Django user will get cleaned up when the test transaction rolls back (because it's in SQL, not Couch).